Dnn.AdminExperience/Library/Dnn.PersonaBar.Library/Containers/PersonaBarContainer.cs

```diff
@@ -16,25 +16,34 @@ namespace Dnn.PersonaBar.Library.Containers
     using Dnn.PersonaBar.Library.Controllers;
     using Dnn.PersonaBar.Library.Helper;
     using Dnn.PersonaBar.Library.Model;
+
     using DotNetNuke.Abstractions;
     using DotNetNuke.Application;
     using DotNetNuke.Entities.Host;
     using DotNetNuke.Entities.Portals;
     using DotNetNuke.Services.Personalization;
+
     using Microsoft.Extensions.DependencyInjection;
+
     using Newtonsoft.Json.Linq;
 
     using Globals = DotNetNuke.Common.Globals;
 
     public class PersonaBarContainer : IPersonaBarContainer
     {
         private static IPersonaBarContainer instance;
+        private readonly IPersonaBarController personaBarController;
 
-        public PersonaBarContainer(INavigationManager navigationManager)
+        /// <summary>Initializes a new instance of the <see cref="PersonaBarContainer"/> class.</summary>
+        /// <param name="navigationManager">The navigation manager.</param>
+        /// <param name="personaBarController">The Persona Bar controller.</param>
+        public PersonaBarContainer(INavigationManager navigationManager, IPersonaBarController personaBarController)

```

Comment:
Should we mark the original overload as obsolete ?

Reply:
I've added an obsolete overload